### PR TITLE
RavenDB-18857 Support for suggestion queries in sharding

### PR DESCRIPTION
**Pull request by arekpalinski**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18857/Sharding-Query-Suggest

### Additional description

Support for suggestion queries in sharded database.

Sorting by popularity isn't supported when query is handled by Corax - https://issues.hibernatingrhinos.com/issue/RavenDB-19840/Sharding-Suggest-Handling-of-sorting-suggestions-by-popularity-in-Corax

### Type of change

- Feature impl for sharding

### How risky is the change?

- Low 

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by Contributor

- Tests have been added that prove the fix is effective or that the feature works

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Documents/Sharding/Queries/Suggestions/ShardedSuggestionQueryOperation.cs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+        return result;
+
+        void AddPopularity(BlittableJsonReaderArray popularityArray, AggregatedSuggestions aggregatedSuggestion, SuggestionResult suggestionResult)
+        {
+            var i = 0;
+
+            aggregatedSuggestion.SuggestionsWithPopularity ??= new Dictionary<string, SuggestWord>();
+
+            foreach (BlittableJsonReaderObject pJson in popularityArray.Cast<BlittableJsonReaderObject>())
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Sharding/Queries/Suggestions/ShardedSuggestionQueryOperation.cs:154]: maybe we can deserialize that in one shot?

File: src/Raven.Server/Documents/Sharding/Queries/Suggestions/ShardedSuggestionQueryProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Sharding.Queries.Suggestions;
+
+public class ShardedSuggestionQueryProcessor : AbstractShardedQueryProcessor<ShardedQueryCommand, QueryResult, SuggestionQueryResult>
+{
+    private readonly long? _existingResultEtag;
+
+    private readonly string _raftUniqueRequestId;
+    public ShardedSuggestionQueryProcessor(TransactionOperationContext context, ShardedDatabaseRequestHandler requestHandler, IndexQueryServerSide query, long? existingResultEtag, CancellationToken token) : base(context, requestHandler, query, false, false, token)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Sharding/Queries/Suggestions/ShardedSuggestionQueryProcessor.cs:24]: can we used named parameters for bools? I do not know what `false, false` is

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Documents/Sharding/Queries/Suggestions/ShardedSuggestionQueryOperation.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Raven.Server/Documents/Sharding/Queries/Suggestions/ShardedSuggestionQueryOperation.cs:154]: Handled in https://github.com/ravendb/ravendb/pull/15718

File: src/Raven.Server/Documents/Sharding/Queries/Suggestions/ShardedSuggestionQueryProcessor.cs (hunk 2 above)
Comment (arekpalinski) [reply src/Raven.Server/Documents/Sharding/Queries/Suggestions/ShardedSuggestionQueryProcessor.cs:24]: Handled in https://github.com/ravendb/ravendb/pull/15718

**Pull request merged**